transactions/pendingtxtracker_test.go

```diff
@@ -265,17 +265,29 @@ func TestPendingTxTracker_MultipleClients(t *testing.T) {
 		(*res)["blockNumber"] = transactionSuccessStatus
 	})
 
-	for i := range txs {
-		err := m.TrackPendingTransaction(txs[i].ChainID, txs[i].Hash, txs[i].From, txs[i].Type, true)
-		require.NoError(t, err)
-	}
+	// If we call TrackPendingTransaction immediately, there is a chance that some events
+	// will be emitted before we reach select, so occasionally select fails on timeout.
```

Comment:
The subscribe is done after `TrackingPendingTransaction` to avoid processing the `addPending` events. The fact that we have a timeout means there are fewer events processed than expected.

Reply:
Sorry, I just now realized the base for this PR is not the latest master. I extended the test to validate the implementation expectations in the latest master. Let's monitor the last change and see if that was enough.

> ... is not guaranteed that fetchTransactions is not executed before select and that will cause not enough events

You were right, we might have missed events because the subscription could have happened before `fetchTransactions` started. I fixed this in the latest master by subscribing before `TrackPendingTransaction`.